src/Nethermind/Nethermind.Core/Caching/LruKeyCache.cs

```diff
@@ -100,13 +101,12 @@ public void Delete(TKey key)
 
         private void Replace(TKey key)
         {
-            // TODO: some potential null ref issue here?
-            
             LinkedListNode<TKey>? node = _lruList.First;
             _lruList.RemoveFirst();
-            _cacheMap.Remove(node.Value);
+            if (node != null) _cacheMap.Remove(node.Value);
```

Comment:
This TODO is terrifying.  I went ahead and fixed it (thanks null checking compiler!).  Is this sort of fix acceptable during  cleanup or should it be reserved for actual bug fixes since it is a code change, not just a compiler hint change like most of the stuff in this PR?

Reply:
@LukaszRozmej is this what you fixed or is it before the fix? was it related?
@MicahZoltu is this the old cache

Reply:
🤷  branch is off of `pruning_uat` branch (and PR is against the same).  I only know that it was missing a null check that would result in an NPE if `node` was ever null (which I believe will happen if the linked list is empty for any reason).